src/compose/application-manager.ts

```diff
@@ -213,6 +214,10 @@ export async function inferNextSteps(
 		} else {
 			steps.push({ action: 'ensureSupervisorNetwork' });
 		}
+	} else if (
```

Comment:
The `ensureSupervisorNetwork` step checks for services using the supervisor-api feature and kills them if `supervisorNetworkReady` is false. I'm not sure how it can happen that the supervisor network gets removed while there are container still attached to it. 

I don't think that can happen with volumes (so your code is correct here), but we might want to do some testing